ysql_table_row_count.py

```diff
@@ -0,0 +1,116 @@
+# pip install psycopg2
+
+import threading
+import time
+import random
+import os
+from functools import partial
+import psycopg2
+
+from multiprocessing.dummy import Pool as ThreadPool
+
+# Which cluster? Which schema? How many sub-tasks? Max parallelism?
+schema_name="public"
+num_tasks_per_table=4096
+num_parallel_tasks=32
+
+cluster = psycopg2.connect("host=127.0.0.1 port=5433 dbname=yugabyte user=yugabyte password=yugabyte")
+session = cluster.cursor()
+
+
+threads = []
+
+def check_partition_row_count(schema_name, table_name, p_columns, cursors, l_bound):
+
+  tid = threading.current_thread().ident
+  if tid not in threads:
+  	threads.append(tid)
+  	
+  index = threads.index(tid)
+  stmt = ("SELECT count(*) as rows FROM {} " +
+          "WHERE yb_hash_code({}) >= %s " +
+          "AND yb_hash_code({}) <= %s").format(
+                                                table_name,
+                                                p_columns,
+                                                p_columns)
+
+  range_size = int((64*1024)/num_tasks_per_table)
+  u_bound = l_bound + range_size - 1
+  cursor = cursors[index]
+  results = cursor.execute(stmt, (int(l_bound), int(u_bound)))
+  assert(cursor.rowcount == 1)
+  row_cnt = cursor.fetchall()[0][0]
+  return row_cnt
+
+def check_table_row_counts(schema_name, table_name):
+    print("Checking row counts for: " + schema_name + "." + table_name);
+    results = session.execute("SELECT a.attname " +
```

Comment:
For a COMPOUND PRIMARY KEY situation, this query returns both the HASH partition columns and the CLUSTERING columns in the primary key... but for yb_hash_code purposes we only want eh HASH columns in the Primary Key I believe.

For example, suppose we have this table `txns`:

```
CREATE TABLE txns (
   buyer_id    text,
   network_id  text,
   txn_number  integer,
   txn_details text,
   PRIMARY KEY(buyer_id, txn_number));
```

then the query you have here:

```
yugabyte=# select a.attname FROM pg_index i JOIN pg_attribute a ON a.attrelid = i.indrelid AND a.attnum = ANY(i.indkey) WHERE i.indrelid = 'txns'::regclass AND i.indisprimary;
  attname
------------
 buyer_id
 txn_number
```

returns you both columns from the PRIMARY KEY:

  PRIMARY KEY(buyer_id, txn_number));

but in this case we want only the `buyer_id` which is the HASH column.